### PR TITLE
Add extended test for build from private source repository

### PR DESCRIPTION
**Pull request by csrwng**
Uses the internal gitserver to test a build with a custom ca-cert and username/password.

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees @mfojtik ptal

**Comment by csrwng** [comment]
[test][extended:core]

**Comment by bparees** [comment]
lgtm

File: test/extended/builds/gitauth.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	testutil "github.com/openshift/origin/test/util"
+)
+
+// hostName removes a port from a host specification
+func hostName(host string) string {
+	parts := strings.Split(host, ":")
+	return parts[0]
```
Comment (mfojtik) [inline test/extended/builds/gitauth.go]: this can be one-liner:

```
return host[0:strings.Index(":")]
```

Comment (mfojtik) [inline test/extended/builds/gitauth.go]: also i would call this function hostname() ;-)

**Comment by mfojtik** [comment]
one small nit, LGTM

File: test/extended/builds/gitauth.go (hunk 1 above)
Comment (liggitt) [inline test/extended/builds/gitauth.go]: This should use net.SplitHostPort and handle cases where no port was part of the host string

**Comment by csrwng** [comment]
[test][extended:core]

File: test/extended/builds/gitauth.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			g.By(fmt.Sprintf("calling oc new-app -f %q -p ROUTE_SUFFIX=%s", gitServerFixture, routeSuffix))
+			err = oc.Run("new-app").Args("-f", gitServerFixture, "-p", fmt.Sprintf("ROUTE_SUFFIX=%s", routeSuffix)).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("expecting the deployment of the gitserver to be in the Complete phase")
+			err = exutil.WaitForADeployment(oc.KubeREST().ReplicationControllers(oc.Namespace()), gitServerDeploymentConfigName,
+				exutil.CheckDeploymentCompletedFunc, exutil.CheckDeploymentFailedFunc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By(fmt.Sprintf("creating a new secret for the gitserver by calling oc secrets new-basicauth --username=%s --password=%s --cacert=%s",
```
Comment (jhadvig) [inline test/extended/builds/gitauth.go]: just a smallest thing: you are missing the secretName ... oc secrets new-basicauth secretName ...

**Comment by jhadvig** [comment]
Just that one small nit, LGTM

**Comment by csrwng** [comment]
thx, fixed
[test][extended:core]

**Comment by jhadvig** [comment]
@csrwng please rebase :-)

**Comment by csrwng** [comment]
[test][extended:core(builds:)]

**Comment by csrwng** [comment]
[testonlyextended][extended:core(gitauth)]

**Comment by deads2k** [comment]
jenkins issue re[test]

**Comment by mfojtik** [comment]
LGTM [merge]

**Comment by csrwng** [comment]
[merge]

**Pull request merged**